supportingFunctions.R

```diff
@@ -0,0 +1,118 @@
+##supporting functions for analysis
+
+#Text file format conversion function
+#This function converts all txt files in a given directory to csv files
+#usage: txt_convert("./countryX")
+#first define a function to convert a single txt file
+single_txt_convert <- function(file_name){
+  data <- read.table(file_name, sep = " ", header = TRUE)
+  new_file_name <- gsub('txt', 'csv', file_name)
+  write.csv(data, file = new_file_name, row.names=F)
+}
+#then use the function above to define the function for converting multiple files
+txt_convert <- function(file_path){
+  file_name_list <- list.files(path = file_path, pattern = ".txt")
+  for(file in file_name_list){
+    name_including_path <- paste(file_path, file, sep = "/")
+    single_txt_convert(name_including_path)
+  }
+}
+
+#Compilation function
+#This function compiles all data of a given list of countries. Users should input a vector containing the name(s) of one or multiple countries.
+#usage: all_data <- compile(c("X","Y"), NA_remove = FALSE, NA_warning = FALSE)
+#requirements: the folders containing data from different countries should be put in the working directory, and they should be named like "countryX"
+#define a function to add columns of country and date to a dataframe
+add_col <- function(dataframe, country, date){
+  country_col <- rep(country, times = nrow(dataframe))
+  date_col <- rep(date, times = nrow(dataframe))
+  country_col <- data.frame(country_col)
+  date_col <- data.frame(date_col)
+  data_to_add <- cbind(country_col, date_col)
+  colnames(data_to_add) <- c("country","dayofYear")
+  dataframe <- cbind(dataframe, data_to_add)

```

Comment:
lines 27 - 33 can be shorten:
dataframe$country = country
dataframe$dayofYear = date
return(dataframe)

Reply:
+2